cfpb/consumerfinance.gov
# Update the header state for handling overlay show/hide logic

## Additions

- Adds boolean variables to track menu and global search open state.
- Adds functions to show/hide overlay.

## Changes

- Makes overlay show when either mega menu or search are open.


## How to test this PR

1. Check menu/search states at mobile:
 - Open menu = overlay should show, search should close if open.
 - Open search = overlay should show, menu should close if open.
 - Close menu = overlay should hide, if search is not opened.
 - Close search = overlay should hide, if menu is not opened.
 - Click overlay = menu or search should close and overlay should hide.
